src/main/java/dev/jbang/RunContext.java

```diff
@@ -0,0 +1,281 @@
+	}
+
+	public void setArguments(List<String> arguments) {
+		this.arguments = arguments;
+	}
+
+	public Map<String, String> getProperties() {
+		return (properties != null) ? properties : Collections.emptyMap();
+	}
+
+	public void setProperties(Map<String, String> properties) {
+		this.properties = properties;
+	}
+
+	public List<String> getAdditionalDependencies() {
+		return additionalDeps;
+	}
+
+	public void setAdditionalDependencies(List<String> deps) {
+		if (deps != null) {
+			this.additionalDeps = new ArrayList<>(deps);
+		} else {
+			this.additionalDeps = Collections.emptyList();
+		}
+	}
+
+	public List<String> getAdditionalClasspaths() {
+		return additionalClasspaths;
+	}
+
+	public void setAdditionalClasspaths(List<String> cps) {
+		if (cps != null) {
+			this.additionalClasspaths = new ArrayList<>(cps);
+		} else {
+			this.additionalClasspaths = Collections.emptyList();
+		}
+	}
+
+	/**
+	 * Returns the Alias object if originalRef is an alias, otherwise null
+	 */
+	public AliasUtil.Alias getAlias() {
+		return alias;
+	}
+
+	/**
+	 * Sets the Alias object if originalRef is an alias
+	 */
+	public void setAlias(AliasUtil.Alias alias) {
+		this.alias = alias;
+	}
+
+	public boolean isForceJsh() {
+		return forceJsh;
+	}
+
+	public void setForceJsh(boolean forcejsh) {
+		this.forceJsh = forcejsh;
+	}
+
+	/**
+	 * The original script reference. Might ba a URL or an alias.
+	 */
+	public String getOriginalRef() {
+		return originalRef;
+	}
+
+	public void setOriginalRef(String ref) {
+		this.originalRef = ref;
+	}
+
+	public String getMainClass() {
+		return mainClass;
+	}
+
+	public String getMainClassOr(Source src) {
+		return (mainClass != null) ? mainClass : src.getMainClass();
+	}
+
+	public void setMainClass(String mainClass) {
+		this.mainClass = mainClass;
+	}
+
+	public List<String> getRuntimeOptions() {
+		return javaRuntimeOptions;
+	}
+
+	public List<String> getRuntimeOptionsOr(Source src) {
+		return (javaRuntimeOptions != null) ? javaRuntimeOptions : src.getRuntimeOptions();
+	}
+
+	public void setRuntimeOptions(List<String> javaRuntimeOptions) {
+		this.javaRuntimeOptions = javaRuntimeOptions;
+	}
+
+	public List<String> getPersistentJvmArgs() {
+		return persistentJvmArgs;
+	}
+
+	public void setPersistentJvmArgs(List<String> persistentJvmArgs) {
+		this.persistentJvmArgs = persistentJvmArgs;
+	}
+
+	public int getBuildJdk() {
+		return buildJdk;
+	}
+
+	public void setBuildJdk(int javaVersion) {
+		this.buildJdk = javaVersion;
+	}
+
+	public String getJavaAgentOption() {
+		return javaAgentOption;
+	}
+
+	public void setJavaAgentOption(String option) {
+		this.javaAgentOption = option;
+	}
+
+	public String getAgentMainClass() {
+		return agentMainClass;
+	}
+
+	public void setAgentMainClass(String b) {
+		agentMainClass = b;
+	}
+
+	public String getPreMainClass() {
+		return preMainClass;
+	}
+
+	public void setPreMainClass(String name) {
+		preMainClass = name;
+	}
+
+	public class SourceContext {
```

Comment:
what is this thing? ..and should it not be static?

Reply:
iread it as a AgentContext where source is representing an agent and then the runcontext for that agent ?  is that about right?